### PR TITLE
Simplify coder creation

### PR DESCRIPTION
**Pull request by davesque**
### What was wrong?

The issue was discussed in ticket #53 .  The coder creation process seems a bit complicated.

### How was it fixed?

Change coder classes to use simple instantiation instead of on-the-fly subclassing to create coder callables.

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

**Comment by davesque** [comment]
Here's a first few steps in the direction that was discussed in the issue ticket.  Let me know if it looks decent.

File: eth_abi/decoding.py [hunk 1]
```diff
@@ -24,28 +24,11 @@
...
-            name = cls.__name__
-        sub_cls = type(name, (cls,), kwargs)
-        sub_cls.validate()
-        instance = sub_cls()
-        return instance
-
-    @classmethod
-    def validate(cls):
-        pass
+    def decode(self, stream):  # pragma: no cover
```
Comment (pipermerriam) [inline eth_abi/decoding.py]: Now that we're instantiating these, maybe we should make use of `abc.ABCMeta` and `abc.abstractmethod`?

**Review by pipermerriam — APPROVED** [review]

**Review by carver — COMMENTED** [review]
Looks good :+1:

File: eth_abi/decoding.py (hunk 1 above)
Comment (davesque) [reply eth_abi/decoding.py]: Sure.  Made that change.

**Pull request merged**